src/vm/vm_core.rs

```diff
@@ -4015,4 +4134,90 @@ mod tests {
+            .run_finished(true)
+            .current_step(12)
+            ._program_base(Some(MaybeRelocatable::from((1, 0))))
+            .accessed_addresses(Some(vec![Relocatable::from((1, 0))]))
+            .builtin_runners(vec![(
+                "string",
+                BuiltinRunner::from(HashBuiltinRunner::new(12, true)),
+            )])
+            .run_context(RunContext {
+                pc: Relocatable::from((0, 0)),
+                ap: 18,
+                fp: 0,
+            })
+            .segments(MemorySegmentManager {
+                segment_sizes: HashMap::new(),
+                segment_used_sizes: Some(vec![1]),
+                public_memory_offsets: HashMap::new(),
+                memory: Memory::new(),
+            })
+            .skip_instruction_execution(true)
+            .trace(Some(vec![TraceEntry {
+                pc: Relocatable::from((0, 1)),
+                ap: Relocatable::from((0, 1)),
+                fp: Relocatable::from((0, 1)),
+            }]))
+            .build();
+        assert!(virtual_machine_from_builder.run_finished);
+        assert_eq!(virtual_machine_from_builder.current_step, 12);
+        assert_eq!(
+            virtual_machine_from_builder._program_base,
+            Some(MaybeRelocatable::from((1, 0)))
+        );
+        assert_eq!(
+            virtual_machine_from_builder.accessed_addresses,
+            Some(vec![Relocatable::from((1, 0))])
+        );
+        assert_eq!(
+            virtual_machine_from_builder
+                .builtin_runners
+                .get(0)
+                .unwrap()
+                .0,
+            "string"
+        );
+        assert_eq!(virtual_machine_from_builder.run_context.ap, 18,);
+        assert_eq!(
+            virtual_machine_from_builder.segments.segment_used_sizes,
+            Some(vec![1])
+        );
+        assert!(virtual_machine_from_builder.skip_instruction_execution,);
+        assert_eq!(
+            virtual_machine_from_builder.trace,
+            Some(vec![TraceEntry {
+                pc: Relocatable::from((0, 1)),
+                ap: Relocatable::from((0, 1)),
+                fp: Relocatable::from((0, 1)),
+            }])
+        );
+
+        #[cfg(feature = "hooks")]
+        use std::sync::Arc;
+
+        #[cfg(feature = "hooks")]
+        fn pre_step_hook(
+            _vm: &mut VirtualMachine,
+            _hint_processor: &mut dyn HintProcessor,
+            _exec_scope: &mut ExecutionScopes,
+            _hint_data: &HashMap<usize, Vec<Box<dyn Any>>>,
+            _constants: &HashMap<String, Felt>,
+        ) -> Result<(), VirtualMachineError> {
+            Ok(())
+        }
+
+        #[cfg(feature = "hooks")]
+        let _virtual_machine_from_builder: VirtualMachine = VirtualMachineBuilder::default()
+            .hooks(crate::vm::hooks::Hooks::new(
+                None,
+                Some(Arc::new(pre_step_hook)),
```

Comment:
`Some(std::sync::Arc::new(pre_step_hook)),` will spare you the import of line 4200